packages/datadog-instrumentations/src/passport-utils.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { channel } = require('./helpers/instrument')
+const passportVerifyChannel = channel('datadog:passport:verify:finish')
+
+function wrapVerifiedAndPublish (username, password, verified, strategy) {
+  if (passportVerifyChannel.hasSubscribers) {
+    return shimmer.wrap(verified, function (err, user, info) {
+      const credentials = { type: strategy, username }
+      passportVerifyChannel.publish({ credentials, user })
+      return verified.call(this, err, user, info)
+    })
+  } else {
+    return verified
+  }
```

Comment:
In js we usually use short circuit syntax, it's more idiomatic like this:
```
if (!passwordVerifyChannel.hasSubscribers) return verified

return shimmer.wrap...
```
but that's just my 2 cents, you can keep it like that if you want